src/redis_request.cc

```diff
@@ -59,11 +59,15 @@ Status Request::Tokenize(evbuffer *input) {
         svr_->stats_.IncrInbondBytes(len);
         if (line[0] == '*') {
           try {
-            multi_bulk_len_ = std::stoull(std::string(line + 1, len-1));
+            multi_bulk_len_ = std::stoll(std::string(line + 1, len-1));
           } catch (std::exception &e) {
             free(line);
             return Status(Status::NotOK, "Protocol error: invalid multibulk length");
           }
+          if (multi_bulk_len_ <= 0) {
```

Comment:
Is a `multi_bulk_len_ = 0` needed while `multi_bulk_len_ < 0` ?

Reply:
There is a condition `if( multi_bulk_len_ == 0)` in `src/redis_request.cc:111`

Reply:
`state_ `is still `ArrayLen`, so it will read again and assign to `multi_bulk_len_`.   it's a good habit that to add the assignment `multi_bulk_len_`. Thank you